pkg/server/api/svid/v1/service.go

```diff
@@ -324,6 +328,96 @@ func (s *Service) NewJWTSVID(ctx context.Context, req *svid.NewJWTSVIDRequest) (
 	}, nil
 }
 
-func (s *Service) NewDownstreamX509CA(context.Context, *svid.NewDownstreamX509CARequest) (*svid.NewDownstreamX509CAResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (s *Service) newX509CASVID(ctx context.Context, entry *types.Entry, csrBytes []byte) (resp *svid.NewDownstreamX509CAResponse, err error) {
+	log := rpccontext.Logger(ctx)
+
+	if !entry.Downstream {
+		err := status.Errorf(codes.NotFound, "error not found or not authorized")
+		log.WithError(err).Error("Rejecting NewDownstreamX509CA request due to downstream flag not being set")
```

Comment:
After thinking about it -- I changed this error to "error not found or not authorized" because I don't necessarily want to reveal to the outside world exactly why it isn't authorized. What do you think?